tm2/pkg/telemetry/metrics/metrics.go

```diff
@@ -112,14 +114,35 @@ var (
 )
 
 func Init(config config.Config) error {
-	// Use oltp metric exporter
-	exp, err := otlpmetricgrpc.New(
-		context.Background(),
-		otlpmetricgrpc.WithEndpoint(config.ExporterEndpoint),
-		otlpmetricgrpc.WithInsecure(),
+	var (
+		ctx = context.Background()
+		exp sdkMetric.Exporter
 	)
+
+	u, err := url.Parse(config.ExporterEndpoint)
 	if err != nil {
-		return fmt.Errorf("unable to create metrics exporter, %w", err)
+		return fmt.Errorf("error parsing exporter endpoint: %s, %w", config.ExporterEndpoint, err)
+	}
+
+	// Use oltp metric exporter with http/https or grpc
+	switch u.Scheme {
+	case "http", "https":
+		exp, err = otlpmetrichttp.New(
+			ctx,
+			otlpmetrichttp.WithEndpointURL(config.ExporterEndpoint),
```

Comment:
Will this work for https if the `WithInsecure` option is not set here?

Reply:
I had the same concerned, i tried locally with my local prometheus and with production one with https and both works